plum/type.py

```diff
@@ -229,7 +229,10 @@ def __hash__(self):
         return hash(ptype(self.resolve()))
 
     def __repr__(self):
-        return repr(ptype(self.resolve()))
+        if self.resolved:
+            return repr(ptype(self.resolve()))
+        else:
+            return "ResolvableType(unresolved)"
```

Comment:
```suggestion
            return "<ResolvableType: unresolved>"
```

I like to give an angle-bracket representation back if the output of `__repr__` isn't able to fully reconstruct the object.